examples/dft.masm

```diff
@@ -0,0 +1,288 @@
+    exp                                         # stack: [TWO_ADIC_ROOT_OF_UNITY^(2^(TWO_ADICITY - log2_n))]
+end
+
+#! Computes the kth element of the "frequency domain" (i.e. of the transformed list).
+#!
+#! Input: [k, root_of_unity, n, v_0, ..., v_{n-1}, ...]
+#! Output: [f_k, ...]
+#!
+#! Locals
+#! 0: k
+#! 1: root_of_unity
+#! 2: n (length of the list)
+#! 3: j (counter)
+#! 4: result (partial result of the computation)
+proc.f_k.5
+    loc_store.0                                # store k
+    loc_store.1                                # store root_of_unity
+    loc_store.2                                # store n
+    push.0                                     # Prepare j = 0
+    loc_store.3                                # Set j = 0
+    push.0                                     # Prepare result = 0
+    loc_store.4                                # Set result = 0
+
+    # Stack here: [v0, ..., v{n-1}, ...]
+
+    push.1                                     # push 1 to enter the loop
+    while.true
+        # compute `root_of_unity^(jk)`
+        loc_load.1                             # stack: [root_of_unity, v{j}, ..., v{n-1}]
+        loc_load.0                             # stack: [k , root_of_unity, v{j}, ..., v{n-1}]
+        loc_load.3                             # stack: [j, k , root_of_unity, v{j}, ..., v{n-1}]
+        mul                                    # stack: [j*k , root_of_unity, v{j}, ..., v{n-1}]
+        exp                                    # stack: [root_of_unity^(jk), v{j}, ..., v{n-1}]
+        mul                                    # stack: [v{j} * root_of_unity^(jk), v{j+1}, ..., v{n-1}]
+
+        # update result
+        loc_load.4                             # stack: [result, v{j} * root_of_unity^(jk), v{j+1}, ..., v{n-1}]
+        add                                    # stack: [result + v{j} * root_of_unity^(jk), v{j+1}, ..., v{n-1}]
+        loc_store.4                            # stack: [v{j+1}, ..., v{n-1}]
+
+        # Increment j
+        loc_load.3                             # stack: [j, v{j+1}, ..., v{n-1}]
+        add.1                                  # stack: [j+1, v{j+1}, ..., v{n-1}]
+        dup                                    # stack: [j+1, j+1, v{j+1}, ..., v{n-1}]
+        loc_store.3                            # stack: [j+1, v{j+1}, ..., v{n-1}]
+
+        # Check if we're done looping
+        loc_load.2                             # stack: [n, j+1, v{j+1}, ..., v{n-1}]
+        neq                                    # stack: [1 or 0, v{j+1}, ..., v{n-1}]
+    end
+
+    # Return result
+    loc_load.4                                 # load result
+end
+
+# Stores an array to `INPUT_ARR_LOC`, where an "array" is a list of elements prefixed by their length.
+# The array is stored in the same order, that is [n, v0, ..., v{n-1}].
+# The inverse operation is `proc.retrieve_array`.
+# Input: [n, v0, ..., v{n-1}, ...]
+# Output: [...]
+proc.store_array
+    # Store n
```

Comment:
This `store_array` procedure was written to be self-contained (modulo the fact that it uses the global `INPUT_ARR_LOC` instead of receiving that address on the stack, but I got lazy); i.e. it can be used to store array of any length. So you could do (pseudocode)

```
# Storing arrays of different lengths
exec.store_array([4,1,2,3,4])
exec.store_array([3,1,2,3])
```

Basically, an array is `[length, ele_0, ..., ele_{length-1}]`, and this `mem_store` stores the `length` (which could be different from call to call).